### PR TITLE
Add regression test for #8554

### LINKED ISSUE #8554: Crash using `isinstance` + `super` due to Astroid
### Bug description

When parsing the following ``a.py``:

```python
class A:
    def __init__(self):
        if not isinstance(super(), float):
            pass

```

there is a crash. Note however that this snippet is fine:

```python
class A:
    def __init__(self):
        if isinstance(super(), float):
            pass

```
with `not` removed. Also I believe it's nothing to do with `float`: you can replace this with a custom class, or another built-in.

### Command used

```shell
pylint a.py
```

### Pylint output

<details open>
    <summary>
        pylint crashed with a ``AstroidError`` and with the following stacktrace:
    </summary>

```python
Traceback (most recent call last):
  File "/home/bugreport/.venv/lib/python3.10/site-packages/astroid/inference_tip.py", line 33, in _inference_tip_cached
    result = _cache[func, node]
KeyError: (<function register_builtin_transform.<locals>._transform_wrapper at 0x7fbe7b0e5b40>, <Call l.3 at 0x7fbe7a87c910>)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 772, in _lint_file
    check_astroid_module(module)
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 1001, in check_astroid_module
    retval = self._check_astroid_module(
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 1047, in _check_astroid_module
    walker.walk(node)
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 94, in walk
    self.walk(child)
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 94, in walk
    self.walk(child)
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 94, in walk
    self.walk(child)
  [Previous line repeated 1 more time]
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 91, in walk
    callback(astroid)
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/pylint/checkers/typecheck.py", line 1903, in visit_unaryop
    for error in node.type_errors():
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/astroid/nodes/node_classes.py", line 4466, in type_errors
    return [
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/astroid/nodes/node_classes.py", line 4466, in <listcomp>
    return [
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/astroid/inference.py", line 565, in _infer_unaryop
    for operand in self.operand.infer(context):
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/astroid/nodes/node_ng.py", line 161, in infer
    results = list(self._explicit_inference(self, context, **kwargs))
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/astroid/inference_tip.py", line 40, in _inference_tip_cached
    result = _cache[func, node] = list(func(*args, **kwargs))
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/astroid/brain/brain_builtin_inference.py", line 175, in _transform_wrapper
    result = transform(node, context=context)
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/astroid/brain/brain_builtin_inference.py", line 725, in infer_isinstance
    isinstance_bool = helpers.object_isinstance(obj_node, class_container, context)
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/astroid/helpers.py", line 133, in object_isinstance
    obj_type = object_type(node, context)
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/astroid/helpers.py", line 87, in object_type
    types = set(_object_type(node, context))
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/astroid/helpers.py", line 71, in _object_type
    raise AssertionError(f"We don't handle {type(inferred)} currently")
AssertionError: We don't handle <class 'astroid.objects.Super'> currently

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 736, in _lint_files
    self._lint_file(fileitem, module, check_astroid_module)
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 774, in _lint_file
    raise astroid.AstroidError from e
astroid.exceptions.AstroidError

```

</details>

### Expected behaviour

No crash.

### Pylint version

```shell
pylint 3.0.0a6
astroid 2.15.1
Python 3.10.8 (main, Dec 19 2022, 16:27:32) [GCC 9.4.0]
```

### OS / Environment

linux (Linux)

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Closes #8554

Depends on https://github.com/pylint-dev/astroid/pull/2177

### CODE REVIEW

**Comment by Pierre-Sassoulas** [comment]
This one is going to be technical, here we need astroid 3.0.0a3 but the backport will need 2.15.5.

**Comment by Pierre-Sassoulas** [comment]
Upgrading astroid in #8685 because updating the version is not the only thing to do.

**Comment by Pierre-Sassoulas** [comment]
(So we can backport)

File: doc/whatsnew/fragments/8554.bugfix [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Fixed crash when an argument-less call to `super()` was placed after an operator (e.g. ``not``).
```
Comment (jacobtylerwalls) [inline doc/whatsnew/fragments/8554.bugfix:1]:
```suggestion
Fixed crash when a call to ``super()`` was placed after an operator (e.g. ``not``).
```

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**